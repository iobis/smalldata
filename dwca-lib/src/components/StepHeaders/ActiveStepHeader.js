import PropTypes from 'prop-types'
import React from 'react'
import StepFooter from './StepFooter'
import StepHeader from './StepHeader'

export default function ActiveStepHeader({
  activeStepIndex,
  children,
  nextStep,
  onContinueButtonClick,
  onCopyPreviousDataClick,
  onStepTitleClick,
  stepDescription,
  stepTitle,
  totalSteps
}) {
  const footerVisible = activeStepIndex <= totalSteps
  return (
    <>
      <StepHeader
        className="active has-background-grey has-text-white"
        dataDescription=""
        iconVisible={false}
        onStepTitleClick={onStepTitleClick}
        scrollOnRender={true}
        selectedData=""
        stepDescription={stepDescription}
        stepTitle={stepTitle}>
        {children}
      </StepHeader>
      {footerVisible && (
        <StepFooter
          activeStepIndex={activeStepIndex}
          nextStep={nextStep}
          onContinueButtonClick={onContinueButtonClick}
          onCopyPreviousDataClick={onCopyPreviousDataClick}/>
      )}
    </>
  )
}

ActiveStepHeader.propTypes = {
  activeStepIndex:         PropTypes.number.isRequired,
  children:                PropTypes.node,
  nextStep:                PropTypes.string,
  onContinueButtonClick:   PropTypes.func.isRequired,
  onCopyPreviousDataClick: PropTypes.func,
  onStepTitleClick:        PropTypes.func.isRequired,
  stepDescription:         PropTypes.string.isRequired,
  stepTitle:               PropTypes.string.isRequired,
  totalSteps:              PropTypes.number.isRequired
}
