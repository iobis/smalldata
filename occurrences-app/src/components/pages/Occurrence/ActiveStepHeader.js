import PropTypes from 'prop-types'
import React from 'react'
import StepHeader from './StepHeader'

export default function ActiveStepHeader({ stepDescription, stepTitle }) {
  return (
    <StepHeader
      className="active has-background-grey has-text-white"
      dataDescription=""
      iconVisible={false}
      selectedData=""
      stepDescription={stepDescription}
      stepTitle={stepTitle}/>
  )
}

ActiveStepHeader.propTypes = {
  stepDescription: PropTypes.string.isRequired,
  stepTitle:       PropTypes.string.isRequired
}
