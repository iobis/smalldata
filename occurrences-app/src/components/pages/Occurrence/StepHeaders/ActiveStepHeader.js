import PropTypes from 'prop-types'
import React from 'react'
import StepHeader from './StepHeader'

export default function ActiveStepHeader({ children, onStepTitleClick, stepDescription, stepTitle }) {
  return (
    <StepHeader
      className="active has-background-grey has-text-white"
      dataDescription=""
      iconVisible={false}
      onStepTitleClick={onStepTitleClick}
      selectedData=""
      stepDescription={stepDescription}
      stepTitle={stepTitle}>
      {children}
    </StepHeader>
  )
}

ActiveStepHeader.propTypes = {
  children:         PropTypes.node,
  onStepTitleClick: PropTypes.func.isRequired,
  stepDescription:  PropTypes.string.isRequired,
  stepTitle:        PropTypes.string.isRequired
}
