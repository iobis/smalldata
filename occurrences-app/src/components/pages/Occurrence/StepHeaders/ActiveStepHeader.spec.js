import ActiveStepHeader from './ActiveStepHeader'
import React from 'react'
import { mount } from 'enzyme'

describe('ActiveStepHeader', () => {
  it('renders StepFooter for 0 step with empty copy-previous-data placeholder', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={0}
        nextStep="next step"
        onContinueButtonClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(true)
  })

  it('renders StepFooter with all data if next step is provided', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={1}
        nextStep="next step"
        onContinueButtonClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('copy data from previous entry')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(true)
  })

  it('does not renders StepFooter if next step is not provided', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={1}
        onContinueButtonClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('copy data from previous entry')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(false)
  })

  it('does not renders StepFooter if last step', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={6}
        nextStep="next step"
        onContinueButtonClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(false)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(false)
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(false)
  })
})
