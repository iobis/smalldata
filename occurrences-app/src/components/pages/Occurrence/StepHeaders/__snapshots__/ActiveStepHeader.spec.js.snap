// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveStepHeader renders correctly 1`] = `
<div
  className="step-header columns is-vcentered active has-background-grey has-text-white"
  onClick={[Function]}
>
  <div
    className="column is-1 is-hidden-mobile"
  >
    <p
      className="is-size-5 is-uppercase"
    >
      <b>
        
      </b>
    </p>
  </div>
  <div
    className="column is-7 is-size-5 is-hidden-mobile"
  >
    
  </div>
  <div
    className="column details"
  >
    <div
      className="is-size-6"
    >
      step description
    </div>
    <div
      className="is-size-5 is-uppercase"
    >
      <b
        className="step-title"
      >
        step title
      </b>
    </div>
  </div>
</div>
`;
